cache/lru.go

```diff
@@ -134,3 +134,8 @@ func (l *LRUCache) Fetch(ctx context.Context, keys []string, opts ...Option) (re
 func (l *LRUCache) Name() string {
 	return "in-memory-" + l.name
 }
+
+func (l *LRUCache) Delete(ctx context.Context, key string) error {
+	l.lru.Remove(key)
```

Comment:
You need to take the `mtx` while accessing the `lru`. Release it before calling `l.c.Delete()`.